pulsar-client/src/main/java/org/apache/pulsar/client/impl/ProducerImpl.java

```diff
@@ -1947,8 +1933,34 @@ protected void triggerFlush() {
         }
     }
 
+    // must acquire semaphore before calling
+    private void maybeTriggerBatchTimerTask() {
+        if (this.batchTimerTask == null) {
+            this.batchTimerTask = cnx().ctx().executor()
+                    .schedule(catchingAndLoggingThrowables(() -> {
+                        if (log.isTraceEnabled()) {
+                            log.trace("[{}] [{}] Batching the messages from the batch container from timer thread",
+                                    topic, producerName);
+                        }
+                        // No need to prematurely cut batches if we're not ready to send them.
+                        if (getState() != State.Ready) {
+                            return;
```

Comment:
I think we need to ensure that we schedule or trigger `batchMessageAndSend()` once the producer gets reconnected, otherwise it might not get re-triggered until new `send()` are issued. It could leave the existing incomplete batch for indefinite time.

Reply:
Great point, thanks for catching this case.

Reply:
I handle this case now. I still need to add a test to cover this case, though.